ui/component/sideNavigation/view.jsx

```diff
@@ -250,8 +252,20 @@ function SideNavigation(props: Props) {
   const showSubscriptionSection = shouldRenderLargeMenu && isPersonalized && subscriptions && subscriptions.length > 0;
   const showTagSection = sidebarOpen && isPersonalized && followedTags && followedTags.length;
 
-  let displayedSubscriptions = subscriptions;
-  if (showSubscriptionSection && subscriptions.length > FOLLOWED_ITEM_INITIAL_LIMIT && !expandSubscriptions) {
+  const [rawSubscriptionFilter, setRawSubscriptionFilter] = React.useState('');
+  const subscriptionFilter: string = useDebounce(rawSubscriptionFilter, FILTER_DEBOUNCE_MS);
+
+  const filteredSubscriptions = subscriptions.filter(
```

Comment:
Since `rawSubscriptionFilter` needs to be updated on every character typed (thus causing a render), not sure how the typing performance will be when filtering a huge list (>500 like Tom's) on each character typed.
But at least we are debouncing here, so the renders result in the same output and DOM updates are reduced.

I do see some lag with 100 followings + CPU x4 slowdown (somewhat emulates mobile), but probably ok since it's not a critical action.

Reply:
To be clear, I think we don't need to do this necessarily to ship, but would be good to check if there is any performance impact, especially on slower mobile devices.

Reply:
Can put it up on one of our domains for testing when it's ready.